pkg/controller/actuator_reconcile.go

```diff
@@ -95,6 +95,12 @@ func (a *actuator) Reconcile(ctx context.Context, _ logr.Logger, network *extens
 		}
 	}
 
+	condition := gardencorev1beta1helper.GetCondition(cluster.Shoot.Status.Constraints, "ToDualStackMigration")
+
+	if condition != nil && condition.Status != "DualStackNodesReady" {
+		ipFamilies.Delete(extensionsv1alpha1.IPFamilyIPv6)
+	}
```

Comment:
Nit: You could inline the condition or at least remove the line between variable assignment and `if` block.

```suggestion
	condition := gardencorev1beta1helper.GetCondition(cluster.Shoot.Status.Constraints, "ToDualStackMigration")

	if condition := gardencorev1beta1helper.GetCondition(cluster.Shoot.Status.Constraints, "ToDualStackMigration"); condition != nil && condition.Status != "DualStackNodesReady" {
		ipFamilies.Delete(extensionsv1alpha1.IPFamilyIPv6)
	}
```